addons/website/tools.py

```diff
@@ -8,7 +8,7 @@
 from odoo.tools import DotDict
 
 
-def get_video_embed_code(video_url):
+def get_video_embed_code(video_url, source_with_document_id=False):
```

Comment:
Shouldn't this be another function. Maybe add `get_video_platform_and_id` that returns a tuple like (platform, id, match). Then 2 function `get_video_embed_code` with same behavior than before using `get_video_platform_and_id` and `get_video_thumbnail` also using `get_video_platform_and_id` and were you can put all your parsing. This way we will not have 4 new functions that do pretty much the same in product_image and all the "platform" logic will remain in tools.